examples/bpk-component-dialog/examples.tsx

```diff
@@ -0,0 +1,163 @@
+/*
+ * Backpack - Skyscanner's Design System
+ *
+ * Copyright 2016 Skyscanner Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { useState } from 'react';
+import type { ReactElement, ReactNode } from 'react';
+
+import BpkButton from '../../packages/bpk-component-button';
+import BpkDialog, {
+  HEADER_ICON_TYPES,
+} from '../../packages/bpk-component-dialog';
+import InfoIcon from '../../packages/bpk-component-icon/lg/information-circle';
+import TickIcon from '../../packages/bpk-component-icon/lg/tick';
+import TrashIcon from '../../packages/bpk-component-icon/lg/trash';
+import BpkText, { TEXT_STYLES } from '../../packages/bpk-component-text';
+import { cssModules, withDefaultProps } from '../../packages/bpk-react-utils';
+
+import STYLES from './examples.module.scss';
+
+const getClassName = cssModules(STYLES);
+
+const Paragraph = withDefaultProps(BpkText, {
+  textStyle: TEXT_STYLES.bodyDefault,
+  tagName: 'p',
+  className: getClassName('bpk-dialog-paragraph'),
+});
+
+type Props = {
+  children: ReactElement;
+  dismissible?: boolean;
+  initallyOpen?: boolean;
+  headerIcon?: ReactNode;
+  headerIconType?: (typeof HEADER_ICON_TYPES)[keyof typeof HEADER_ICON_TYPES];
+  flare?: boolean;
+};
+
+const DialogContainer = (props: Props) => {
+  const { children, dismissible = true, initallyOpen = false, ...rest } = props;
+  const [isOpenState, setIsOpen] = useState(initallyOpen);
+  const onOpen = () => {
+    setIsOpen(true);
+  };
+
+  const onClose = () => {
+    setIsOpen(false);
+  };
+
+  return (
+    <div id="dialog-container">
+      <div id="pagewrap">
+        <BpkButton onClick={onOpen}>Open dialog</BpkButton>
+      </div>
+      <BpkDialog
```

Comment:
Can we add some examples (or change the existing ones) to render the dialog within the `pagewrap` element to reproduce how it would actually be like in production and more easily test that it's accessible?